common/djangoapps/student/views.py

```diff
@@ -1699,25 +1702,25 @@ def create_account_with_params(request, params):
         settings.FEATURES.get("ENFORCE_PASSWORD_POLICY", False) and
         not do_external_auth
     )
+    registration_fields = configuration_helpers.get_value('REGISTRATION_EXTRA_FIELDS', getattr(settings, 'REGISTRATION_EXTRA_FIELDS', {}))
+    # No needed so far
     # Can't have terms of service for certain SHIB users, like at Stanford
-    registration_fields = getattr(settings, 'REGISTRATION_EXTRA_FIELDS', {})
-    tos_required = (
-        registration_fields.get('terms_of_service') != 'hidden' or
-        registration_fields.get('honor_code') != 'hidden'
-    ) and (
-        not settings.FEATURES.get("AUTH_USE_SHIB") or
-        not settings.FEATURES.get("SHIB_DISABLE_TOS") or
-        not do_external_auth or
-        not eamap.external_domain.startswith(openedx.core.djangoapps.external_auth.views.SHIBBOLETH_DOMAIN_PREFIX)
-    )
-
+    # tos_required = (
+    #     registration_fields.get('terms_of_service') != 'hidden' or
+    #     registration_fields.get('honor_code') != 'hidden'
+    # ) and (
+    #     not settings.FEATURES.get("AUTH_USE_SHIB") or
+    #     not settings.FEATURES.get("SHIB_DISABLE_TOS") or
+    #     not do_external_auth or
+    #     not eamap.external_domain.startswith(openedx.core.djangoapps.external_auth.views.SHIBBOLETH_DOMAIN_PREFIX)
+    # )
     form = AccountCreationForm(
         data=params,
         extra_fields=extra_fields,
         extended_profile_fields=extended_profile_fields,
         enforce_username_neq_password=True,
         enforce_password_policy=enforce_password_policy,
-        tos_required=tos_required,
+        tos_required=registration_fields.get('terms_of_service') != 'hidden',
```

Comment:
there is no default value, will be 500 when /admin/sites/sites_configuration JSON is not setup correctly

Reply:
...or {} will be used (as last resort option in line 1705)

Reply:
sorry, my bad. .get will return 'None'